src/main/java/com/mathworks/ci/actions/RunMatlabBuildAction.java

```diff
@@ -0,0 +1,100 @@
+
+    private static String DEFAULT_PLUGIN = 
+        "+ciplugins/+jenkins/getDefaultPlugins.m";
+    private static String BUILD_REPORT_PLUGIN = 
+        "+ciplugins/+jenkins/BuildReportPlugin.m";
+    private static String TASK_RUN_PROGRESS_PLUGIN = 
+        "+ciplugins/+jenkins/TaskRunProgressPlugin.m";
+
+    public RunMatlabBuildAction(MatlabCommandRunner runner, BuildConsoleAnnotator annotator, BuildActionParameters params) {
+        this.runner = runner;
+        this.annotator = annotator;
+        this.params = params;
+    }
+
+    public RunMatlabBuildAction(BuildActionParameters params) throws IOException, InterruptedException {
+        this(new MatlabCommandRunner(params), 
+                new BuildConsoleAnnotator(
+                    params.getTaskListener().getLogger(), 
+                    params.getBuild().getCharset()),
+                params);
+    }
+
+    public void run() throws IOException, InterruptedException, MatlabExecutionException {
+        // Copy plugins and override default plugins function
+        runner.copyFileToTempFolder(DEFAULT_PLUGIN, DEFAULT_PLUGIN);
+        runner.copyFileToTempFolder(BUILD_REPORT_PLUGIN, BUILD_REPORT_PLUGIN);
+        runner.copyFileToTempFolder(TASK_RUN_PROGRESS_PLUGIN, TASK_RUN_PROGRESS_PLUGIN);
+        
+        // Set environment variable
+        runner.addEnvironmentVariable(
+                "MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE",
+                "ciplugins.jenkins.getDefaultPlugins");
+
+        // Redirect output to the build annotator
+        runner.redirectStdOut(annotator);
+
+        // Prepare the build tool command
+        // TODO: Devise better solution then prepending the command
+        //   here.
+        String command = "addpath('" 
+            + runner.getTempFolder().getRemote()
+            + "'); buildtool";
+
+        if (params.getTasks() != null) {
+            command += " " + params.getTasks();
+        }
+
+        if (params.getBuildOptions() != null) {
+            command += " " + params.getBuildOptions();
+        }
+
+        try {
+            runner.runMatlabCommand(command);
+        } catch (Exception e) {
+            this.params.getTaskListener().getLogger()
+                .println(e.getMessage());
+            throw(e);
+        } finally {
+            annotator.forceEol();
+        }
+
+        // Handle build result
+        Run<?,?> build = this.params.getBuild();
+        FilePath jsonFile = new FilePath(params.getWorkspace(), ".matlab/buildArtifact.json");
+        if (jsonFile.exists()) {
+            FilePath rootLocation = new FilePath(
+                    new File(
+                        build.getRootDir()
+                        .getAbsolutePath()
+                        + "/buildArtifact.json"));
+            jsonFile.copyTo(rootLocation);
+            jsonFile.delete();
+            build.addAction(new BuildArtifactAction(build, this.params.getWorkspace()));
```

Comment:
Yeah, this is the same behavior that it has in the most recent release which added the build artifact + results action. I think that we can probably rename the file without causing any issues so I will go ahead and update that.

My understanding is similar to yours, since it's using this interface:
https://javadoc.jenkins-ci.org/hudson/model/PersistenceRoot.html

I'll double check though to make sure that no build reports are getting overwritten by confirming where the build root is in practice.

Reply:
No flag however this might introduce compatibility issues with the files which are already archived under root. Considering users have used the recent version of plugin and archives few build results. Any plans on how we will take care of it? or we are safe enough to assume that the impact may not be widers as the release is very recent.

Reply:
Hmm, good point. What are your thoughts @davidbuzinski ? I'm not sure that renaming it is worth the risk of an incompatibility but it seems like if we are going to rename it the sooner we do the better. 

I think I'd prefer to avoid supporting both artifacts named `buildArtifact.json` and `matlabBuildArtifact.json` which is my first thought as to how we would avoid the incompatibility.